infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batch/BatchResult.java

```diff
@@ -0,0 +1,40 @@
+package org.corfudb.infrastructure.log.statetransfer.batch;
+
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import static org.corfudb.infrastructure.log.statetransfer.batch.BatchResult.FailureStatus.SUCCEEDED;
+
+/**
+ * A result of a batch transfer. If completed successfully returns a list of addresses,
+ * status SUCCEEDED and an optional destination server.
+ * If completed exceptionally, returns a list of empty addresses,
+ * status FAILED and an optional destination server.
+ */
+@Builder
+@Getter
+@EqualsAndHashCode
+public class BatchResult {
+
+    public enum FailureStatus {
```

Comment:
Not the same semantics, quite a different domain logic actually. 

I will change the DTO names so that it would make more sense.

- Batch -> TransferBatchRequest (A request to 1. read one batch of addresses either from a destination node (directly/file transfer) or via a protocol and 2. store it in the local stream log). 

- ReadBatch -> ReadBatch (A result of step 1. with all the data needed to handle step 2. or retry step 1.).

- BatchResult -> TransferBatchResponse (A response/result of an entire transfer (step  1. and step 2). It contains only the original transferBatchRequest and a status --> enough data for the upper layer to compute a total result based on it, in case of success, or handle the failures, in case the current layer failed to do so). 

An example would be rescheduling the load on the different destination nodes after all the retries to read from the current destination nodes failed. We don't really need a list of LogData to handle such a failure, what we need is an original request to retry it.